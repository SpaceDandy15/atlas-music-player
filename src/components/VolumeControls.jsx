import React from "react";

export default function VolumeControls() {
  return (
    <div className="flex items-center gap-2 w-48">
      <input
        type="range"
        min="0"
        max="100"
        className="w-full accent-gray-700"
      />
    </div>
  );
}
