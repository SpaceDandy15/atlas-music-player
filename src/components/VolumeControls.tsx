import React from "react";
import { Volume2 } from "lucide-react";

interface VolumeControlsProps {
  volume: number; // 0-100
  setVolume: (v: number) => void;
}

const VolumeControls: React.FC<VolumeControlsProps> = ({ volume, setVolume }) => {
  return (
    <div className="flex items-center gap-2 w-48">
      {/* Volume icon */}
      <Volume2 size={24} className="text-[var(--color-bermuda)]" />

      {/* Slider */}
      <input
        type="range"
        min={0}
        max={100}
        value={volume}
        onChange={(e) => setVolume(Number(e.target.value))}
        className="w-full accent-[var(--color-bermuda)]"
      />
    </div>
  );
};

export default VolumeControls;
