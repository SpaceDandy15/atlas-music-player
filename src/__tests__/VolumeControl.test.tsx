import React from "react";
import renderer from "react-test-renderer";
import { describe, it, expect } from "vitest";
import VolumeControls from "../components/VolumeControls";

describe("VolumeControls Component", () => {
  it("renders correctly with default volume", () => {
    // Snapshot test ensures component renders properly with a given volume
    const tree = renderer
      .create(<VolumeControls volume={50} setVolume={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly at volume 0", () => {
    // Snapshot test checks edge case where volume is at minimum
    const tree = renderer
      .create(<VolumeControls volume={0} setVolume={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly at volume 100", () => {
    // Snapshot test checks edge case where volume is at maximum
    const tree = renderer
      .create(<VolumeControls volume={100} setVolume={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
