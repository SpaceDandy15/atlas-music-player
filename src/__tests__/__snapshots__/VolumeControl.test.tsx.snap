// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls Component > renders correctly at volume 0 1`] = `null`;

exports[`VolumeControls Component > renders correctly at volume 100 1`] = `null`;

exports[`VolumeControls Component > renders correctly with default volume 1`] = `null`;
